### PR TITLE
change sides after a set

### PR DESCRIPTION
**Pull request by timonrey**
The most important change is that the buttons are no longer belong to just one player. The Button can change the player who he is counting for. The sense behind that is that the players can change their sides after a set without taking the buttons with them. With the new method **whichPlayerSide** in the game class we can decide which player on which side of the table is playing. Another important change is that the match counter can just be shown after the score of the won match points was changed and not after every score change.

### CODE REVIEW

File: src/main/java/Game.java [hunk 1]
```diff
@@ -21,13 +21,20 @@ public boolean hasPlayerWonSet(Players playerWhoIsWinning, Players playerWhoIsLo
...
         } else if (playerWhoIsWinning.getSetScore() >= 11 && playerWhoIsLoosing.getSetScore() >= 10) {
-            return hasSomeOneOfThePlayersTwoPointsMore();
+            return hasSomeoneOfPlayersTwoPointsMore();
 
         } else {
             return false;
         }
     }
 
+    public boolean whichPlayerSide(Players playerOne, Players playerTwo) {
```
Comment (PhilippSpo) [inline src/main/java/Game.java]: All this function does is calculating whether the match score is even or not. It technically doesn't say anything about which player is playing on which side of the table.

File: src/main/java/GameMonitor.java [hunk 2]
```diff
@@ -4,30 +4,53 @@
...
 
-    public GameMonitor () {
+    public GameMonitor() {
         this.firstPlayer = new Players();
         this.secondPlayer = new Players();
         this.match = new Game(firstPlayer, secondPlayer);
     }
 
     public void buttonPress(Message message) {
+        if (message.getButtonId().equals("1") && match.whichPlayerSide(firstPlayer, secondPlayer)) {
```
Comment (PhilippSpo) [inline src/main/java/GameMonitor.java]: We can encapsulate the logic of calculating which player has pressed the button into a separate method.

**Review by PhilippSpo — CHANGES_REQUESTED** [review]

File: src/main/java/GameMonitor.java [hunk 3]
```diff
@@ -14,12 +14,16 @@ public GameMonitor() {
     public Players playerWhoPressedButton(Message message) {
 
         if (message.getButtonId().equals("1") && match.isSetScoreEven(firstPlayer, secondPlayer)) {
+            System.out.println("player one pressed");
```
Comment (PhilippSpo) [inline src/main/java/GameMonitor.java]: I think we can remove this logs now 😅

**Review by PhilippSpo — APPROVED** [review]

File: src/main/java/Game.java [hunk 4]
```diff
@@ -12,28 +12,34 @@ public boolean hasSomebodyWon() {
         return (hasPlayerWonSet(playerOne, playerTwo) || hasPlayerWonSet(playerTwo, playerOne));
     }
 
-    public boolean hasSomeOneOfThePlayersTwoPointsMore() {
-        return  (Math.abs(this.playerOne.getSetScore() - this.playerTwo.getSetScore()) == 2);
+    public boolean hasSomeoneOfPlayersTwoPointsMore(Player playerWhoIsWinning, Player playerWhoIsLoosing) {
```
Comment (emmenko) [inline src/main/java/Game.java:15]: Typo: `playerWhoIsLoosing` -> `playerWhoIsLosing`

Comment (emmenko) [reply src/main/java/Game.java:15]: Nitpick: maybe rename the method to `hasOneOfThePlayersTwoPointsMore` (it's better to read)

File: src/main/java/Game.java [hunk 5]
```diff
@@ -12,28 +12,34 @@ public boolean hasSomebodyWon() {
...
         } else if (playerWhoIsWinning.getSetScore() >= 11 && playerWhoIsLoosing.getSetScore() >= 10) {
-            return hasSomeOneOfThePlayersTwoPointsMore();
+            return hasSomeoneOfPlayersTwoPointsMore(playerWhoIsWinning, playerWhoIsLoosing);
 
         } else {
             return false;
         }
     }
 
+    public boolean isSetScoreEven(Player playerOne, Player playerTwo) {
```
Comment (emmenko) [inline src/main/java/Game.java:31]: Shouldn't this be `isMatchScoreEven`? 🤔

File: src/main/java/GameMonitor.java [hunk 6]
```diff
@@ -1,33 +1,53 @@
...
 
-        } else if (message.getButtonId().equals("2")) {
-            match.updateScoreOfPlayer(secondPlayer, message.getActionType());
-            if (match.hasSomebodyWon()) {
-                match.updateMatchScoreOfPlayers();
-            }
+    }
 
+    public void buttonPress(Message message) {
+        Player actingPlayer = this.getPlayer(message);
```
Comment (emmenko) [inline src/main/java/GameMonitor.java:29]: Nitpick: do you mean `activePlayer`? 🤔

**Review by emmenko — CHANGES_REQUESTED** [review]
I left some comments about naming things.

Also there is a `out/production/classes/GameMonitor.class` file that should be ignored.

File: src/main/java/Game.java (hunk 4 above)
Comment (timonrey) [reply src/main/java/Game.java:15]: It will be dome ;)

Comment (timonrey) [reply src/main/java/Game.java:15]: done*

File: src/main/java/Game.java (hunk 5 above)
Comment (timonrey) [reply src/main/java/Game.java:31]: Yes, you are absolutely right!

File: src/main/java/GameMonitor.java (hunk 6 above)
Comment (timonrey) [reply src/main/java/GameMonitor.java:29]: I think "acting" is a better designation. The important thing is to know which player is acting (which player has made a point or wants to update the score) and both are active, aren't they?

**Pull request merged**